src/frontend/js/types/Joanie.ts

```diff
@@ -324,6 +333,11 @@ export interface API {
   products: {
     get(filters?: ResourcesQuery): Promise<Nullable<Product>>;
   };
+  courses: {
+    get<Filters extends ResourcesQuery = CourseFilters>(
```

Comment:
I think it should extends CourseFilters.
```suggestion
    get<Filters extends CourseFilters = CourseFilters>(
```